org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java

```diff
@@ -0,0 +1,362 @@
+		}
+
+		static Predicate<FrameworkEvent> isTypeError() {
+			return e -> isType(FrameworkEvent.ERROR).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeInfo() {
+			return e -> isType(FrameworkEvent.INFO).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypePackagesRefreshed() {
+			return e -> isType(FrameworkEvent.PACKAGES_REFRESHED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStarted() {
+			return e -> isType(FrameworkEvent.STARTED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStartlevelChanged() {
+			return e -> isType(FrameworkEvent.STARTLEVEL_CHANGED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStopped() {
+			return e -> isType(FrameworkEvent.STOPPED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStoppedBootclasspathModified() {
+			return e -> isType(FrameworkEvent.STOPPED_BOOTCLASSPATH_MODIFIED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStoppedUpdate() {
+			return e -> isType(FrameworkEvent.STOPPED_UPDATE).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeWaitTimeout() {
+			return e -> isType(FrameworkEvent.WAIT_TIMEDOUT).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeWarning() {
+			return e -> isType(FrameworkEvent.WARNING).test(e);
+		}
+	}
+
+	class ServiceEvents {
+
+		public static Predicate<Object> isServiceEvent() {
+			return e -> e instanceof ServiceEvent;
+		}
+
+		public static Predicate<Object> isServiceEventAnd(Predicate<ServiceEvent> predicate) {
+			return e -> isServiceEvent().test(e) && predicate.test((ServiceEvent) e);
+		}
+
+		// ServiceEvents
+		public static Predicate<ServiceEvent> isType(final int eventTypeMask) {
+			return e -> (e.getType() & eventTypeMask) != 0;
+		}
+
+		public static Predicate<ServiceEvent> isTypeRegistered() {
+			return e -> isType(ServiceEvent.REGISTERED).test(e);
+		}
+
+		public static Predicate<ServiceEvent> isTypeModified() {
+			return e -> isType(ServiceEvent.MODIFIED).test(e);
+		}
+
+		public static Predicate<ServiceEvent> isTypeModifiedEndmatch() {
+			return e -> isType(ServiceEvent.MODIFIED_ENDMATCH).test(e);
+		}
+
+		public static Predicate<ServiceEvent> isTypeUnregistering() {
+			return e -> isType(ServiceEvent.UNREGISTERING).test(e);
+		}
+
+		public static Predicate<ServiceEvent> hasObjectClass(final Class<?> objectClass) {
+
+			return e -> {
+				Object classes = e.getServiceReference()
+					.getProperty(Constants.OBJECTCLASS);
+
+				if (classes != null && classes instanceof String[]) {
+					return Stream.of((String[]) classes)
+						.filter(Objects::nonNull)
+						.anyMatch(objectClass.getName()::equals);
+				}
+				return false;
+			};
+
+		}
+
+		public static Predicate<ServiceEvent> containServiceProperty(final String key, Object value) {
+			return e -> containsServiceProperties(Dictionaries.dictionaryOf(key, value)).test(e);
+		}
+
+		public static Predicate<ServiceEvent> containsServiceProperties(Map<String, Object> map) {
```

Comment:
The map argument should be `Map<String,?>`. In general, for arguments, we should use the most expansive generic types. All other method arguments need similar changes.

Reply:
using `Map<String,?>`